.github/workflows/contrib.yml

```diff
@@ -0,0 +1,40 @@
+name: Contribution checks
+
+on:
+  push:
+    branches:
+      - main
+      - maint/*
+  pull_request:
+    branches:
+      - main
+      - maint/*
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: pipx run ruff check .
+      - run: pipx run ruff format --diff .
+
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
Can you add a dependabot to keep actions up-to-date? Here's one that uses groups to collect pip and actions dependencies into chunked PRs:

https://github.com/OpenNeuroOrg/openneuro/blob/master/.github/dependabot.yml

Reply:
Done!